src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -0,0 +1,115 @@
+import { supportsETH } from '@shapeshiftoss/hdwallet-core'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useCallback, useState } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import type { Swapper2, TradeQuote2 } from 'lib/swapper/api'
+import { SwapperName } from 'lib/swapper/api'
+import { lifi } from 'lib/swapper/swappers/LifiSwapper/LifiSwapper2'
+import { thorchain } from 'lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper2'
+import { assertUnreachable, isEvmChainAdapter } from 'lib/utils'
+import { selectPortfolioAccountMetadataByAccountId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { TRADE_POLL_INTERVAL_MILLISECONDS } from '../constants'
+import { useAccountIds } from '../useAccountIds'
+import { withFromOrXpub } from './helpers'
+
+export const useTradeExecution = ({
+  swapperName,
+  tradeQuote,
+}: {
+  swapperName: SwapperName
+  tradeQuote: TradeQuote2
+}) => {
+  const [sellTxId, setSellTxId] = useState<string | undefined>()
+  const [buyTxId, setBuyTxId] = useState<string | undefined>()
+  const [message, setMessage] = useState<string | undefined>()
```

Comment:
Can we make this a more dev-friendly vernacular? Perhaps something like `display something something` or `body` ? `message` carries the terminology of messaging, and I had to double check the implementation to confirm this was a message as in, an akschual message

Reply:
posssibly, im mirroring the swapper api here, and `message` is somewhat similar to `Error('my message').message => 'my message'`

Reply:
prefer we get this in and iterate on that async